presto-hive-common/src/main/java/com/facebook/presto/hive/HdfsContext.java

```diff
@@ -32,6 +32,7 @@
     // true if the table already exist in the metastore, false if the table is about to be created in the current transaction
     private final Optional<Boolean> isNewTable;
     private final Optional<String> clientInfo;
+    private final Optional<ConnectorSession> session;
```

Comment:
Which field do we need in `HdfsContext` from `ConnectorSession`? Given identity, source, and queryId are extracted from ConnectorSession, we could either only keep ConnectorSession or put a new field to get the necessary info from ConnectorSession.

Reply:
we need the session properties to control the flow